third_party/amd/lib/TritonAMDGPUTransforms/BlockPingpong.cpp

```diff
@@ -151,35 +151,39 @@ LogicalResult Pingponger::genLocalSlice(OpBuilder &builder, Value v,
                                         int64_t sliceWidth) {
   SmallVector<Operation *> slices;
   SmallVector<Operation *> subviews;
-  auto memDesc = v.getDefiningOp()->getOperand(0);
-  auto type = cast<ttg::MemDescType>(memDesc.getType());
-  SmallVector<int64_t> shape = llvm::to_vector(type.getShape());
-  Type elementType = type.getElementType();
-  int64_t kIdx = opIdx == 0 ? 1 : 0;
-  shape[kIdx] = sliceWidth;
-  // Each slice cannot be smaller than the smallest supported mfma width.
-  if (sliceWidth < 16)
-    return failure();
-  auto dotOperandEnc = ttg::DotOperandEncodingAttr::get(
-      builder.getContext(), opIdx, dotEncoding, kWidth);
-  auto subviewDescType = ttg::MemDescType::get(
-      shape, elementType, type.getEncoding(), type.getMemorySpace(),
-      type.getMutableMemory(), type.getAllocShape());
-  for (int i = 0; i < numSlices; i++) {
-    SmallVector<Value> offsetsVal;
-    SmallVector<int64_t> offsets = {0, 0};
-    offsets[kIdx] = i;
-    for (int64_t off : offsets) {
-      offsetsVal.push_back(constOffsets[off]);
+  // TODO: support transformed input to dot
```

Comment:
Can you add a lit test too? So it will be used for the future when we enable it.

Reply:
good point!

Reply:
Added a test, it can be still used when enabled but we may want to have more practical case such like including upscale. Still not sure which sequence could be supported, will add better tests when actually we enable it.